ZOOKEEPER-3958: Update dependency versions and eliminate java docs wa…

tamaashu opened the pull request:
…rnings

We have quite many outdated dependency which can be updated and also a few warnings during the build of JavaDocs.

ctubbsii commented on zookeeper-server/src/main/java/org/apache/zookeeper/cli/CliCommand.java:
```diff
@@ -72,23 +72,23 @@ public void setZk(ZooKeeper zk) {
     /**
      * get the string used to call this command
-     * @return
+     * @return cmdStr
```
These don't really provide any value to the reader of the javadoc, as this variable is implementation details and are meaningless to the person reading the javadocs. These unneeded return tags should just be removed from the javadoc in these cases where there's no meaningful description in the return tag.

tamaashu replied:
Thanks, fixed.

symat reviewed (approved):
LGTM, thanks for the fixes / dependency upgrades.
(it might worth to run a CVE check, although we only upgraded to newer versions, so I don't expect that we made things worse than they were)

eolivelli reviewed (approved):
LGTM

tamaashu commented:
> LGTM, thanks for the fixes / dependency upgrades.
> (it might worth to run a CVE check, although we only upgraded to newer versions, so I don't expect that we made things worse than they were)

I have updated these dependencies to their latest version, so if they contain any CVE then we have a problem...

tamaashu commented:
@eolivelli I have added a refreshment to dropwizard and prometheus too.
Metrics work for me properly, but since you have added prometheus it might worth to check deeper by you.
I've also updated the documentation as it didn't work for me on the written way with the newest prometheus (v2.21.0).

tamaashu commented:
I've just realized, I have to update some license files too. New commit needed.

symat reviewed (approved):
LGTM, +1

anmolnar reviewed (approved):
lgtm. As long as we only apply this on the master branch.
For stable branches we should have good reasons to upgrade dependencies.

maoling commented:
@tamaashu  The build warnings is an annoying and long-term coexistence issue. Thanks for this.
By executing `mvn clean package -DskipTests`, I found other **61** building warnings(Look at this [file](https://issues.apache.org/jira/secure/attachment/13013603/build_warning.txt) I attached inot the JIRA). Am I missing something?

tamaashu commented:
> @tamaashu The build warnings is an annoying and long-term coexistence issue. Thanks for this.
> By executing `mvn clean package -DskipTests`, I found other **61** building warnings(Look at this [file](https://issues.apache.org/jira/secure/attachment/13013603/build_warning.txt) I attached inot the JIRA). Am I missing something?

@maoling sorry, but after I ran the build with `mvn clean package -DskipTests`, I've only got one warning:
`[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ zookeeper-compatibility-tests-curator ---
[WARNING] JAR will be empty - no content was marked for inclusion!`

Is it possible that you have used an earlier version of this change?

tamaashu commented:
@eolivelli do you think this is ready to commit into the master branch?

eolivelli reviewed (approved):
LGTM

eolivelli commented:
committed, assuming that @ctubbsii comment has been addressed
